boskos/client/client.go

```diff
@@ -640,3 +693,34 @@ func isDialErrorRetriable(err error) bool {
 	}
 	return false
 }
+
+// workFunc describes retrieable work. It should
+// * Return an error for non-recoverable errors
+// * Write retriable errors into `retriedErrs` and return with false, nil
+// * Return with true, nil on success
+type workFunc func(retriedErrs *[]error) (bool, error)
+
+// SleepFunc is called when requests are retried. This may be replaced in tests.
+var SleepFunc = time.Sleep
+
+func retry(work workFunc) error {
+	var retriedErrs []error
+
+	maxAttempts := 4
+	for i := 0; i < maxAttempts; i++ {
```

Comment:
The `+1` and `-1` below can be removed by making the loop index 1-based.

Reply:
Hm yeah, I didn't do it initially because a loop that starts at index one breaks eveyones assumptions, but then again it does make the code much more readable